pennylane/ops/qubit/hamiltonian.py

```diff
@@ -582,21 +582,26 @@ def __sub__(self, H):
 
     def __iadd__(self, H):
         r"""The inplace addition operation between a Hamiltonian and a Hamiltonian/Tensor/Observable."""
+        if not isinstance(H, Observable):
+            raise ValueError(f"Cannot add Hamiltonian and {type(H)}")
+
         if isinstance(H, Hamiltonian):
             self._coeffs = qml.math.concatenate([self._coeffs, H.coeffs], axis=0)
             self._ops.extend(H.ops.copy())
-            self.simplify()
-            return self
 
         if isinstance(H, (Tensor, Observable)):
             self._coeffs = qml.math.concatenate(
                 [self._coeffs, qml.math.cast_like([1.0], self._coeffs)], axis=0
             )
             self._ops.append(H)
-            self.simplify()
-            return self
 
-        raise ValueError(f"Cannot add Hamiltonian and {type(H)}")
+        # HOTFIX: repeat some of the __init__ logic
+        self._hyperparameters = {"ops": self._ops}
+        self._wires = qml.wires.Wires.all_wires([op.wires for op in self.ops], sort=True)
+        self._grouping_indices = None
+        self.data = [self._coeffs[i] for i in range(qml.math.shape(self._coeffs)[0])]
```

Comment:
Ah Yes! We need to update all of the meta data, not just the wires. 
In this case we could get rid of the wires being updated in the `simplify` method as this is now redundant